lib/backend/model/block.go

```diff
@@ -90,13 +90,22 @@ type AllocationBlock struct {
 	Allocations    []*int                `json:"allocations"`
 	Unallocated    []int                 `json:"unallocated"`
 	Attributes     []AllocationAttribute `json:"attributes"`
+	Deleting       bool                  `json:"deleting"`
 
 	// HostAffinity is deprecated in favor of Affinity.
 	// This is only to keep compatiblity with existing deployments.
 	// The data format should be `Affinity: host:hostname` (not `hostAffinity: hostname`).
 	HostAffinity *string `json:"hostAffinity,omitempty"`
 }
 
+func (b *AllocationBlock) MarkDeleted() {
+	b.Deleting = true
```

Comment:
Difference between Deleting and Deleted seems a bit off.  I have a mild preference for standardising on "Deleted"